src/types/jwtDataTypes.ts

```diff
@@ -0,0 +1,94 @@
+import * as z from 'zod';
+
+/* Base JWT types */
+export const BaseJwtData = z
+  .object({
+    iat: z.number(),
+    exp: z.number(),
+    sub: z.string(),
+    iss: z.string(),
+    jti: z.string(),
+  })
+  .strict();
```

Comment:
using strict here will cause this library to reject JWTs if new props are added in the future, this requires tighter coupling between this library and Ego. For that reason I reccomend using the default behaviour (ignores extra props).